winboat: init at 0.8.7

Rexcrazy804 opened the pull request:
Initialized nix derivation for [winboat](https://github.com/TibixDev/winboat)
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
- added rexies to maintainers
- added pkgs/by-name/wi/winboat/package.nix
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Rexcrazy804 committed:
maintainers: add rexies

Rexcrazy804 committed:
winboat: init at 0.8.7

ppom0 reviewed (commented):
Great! After this is merged, we could optionnaly make a `programs.winboat.enable` NixOS option that enables docker and adds winboat to `environment.systemPackages`. Do you see something else that the module should do?

Rexcrazy804 commented on pkgs/by-name/wi/winboat/package.nix:
I've force pushed with the following changes
- seprated derrivations for server and winboat front end
- used `$SOURCE_DATE_EPOCH` as instructed (thanks for this code snippet I appreciate it :)

ppom0 reviewed (approved):
Definitely looks cleaner to use `pkgsCross` and `buildGoModule`. I didn't think of this. Thanks!

Scrumplex commented on pkgs/by-name/wi/winboat/package.nix:
Nit: `passthru` is usually for attributes that shouldn't affect the build/change the derivation. In this case we do want it to affect the derivation, so we could just rename `guestDrv` to `guest-server`.


```suggestion
  guest-server = pkgsCross.mingwW64.callPackage ./guest-server.nix { };
```

Scrumplex reviewed (approved):
This looks good overall. I haven't tested this myself yet

Rexcrazy804 commented:
- renamed `guestDrv` attribute to `guest-server`
- removed `go` package from the derriation (left over from before the guest-server was moved to a separate derivation)

edit:
- removed `writableTmpDirAsHomeHook` same reason as above.

ppom0 reviewed (approved):
Commit https://github.com/NixOS/nixpkgs/commit/4fdc48ee7d9f4d8259579d0b6aff74c3312cc5bb compiles and runs well on my end, well done!

ppom0 commented:
@v3rm1n0 indeed!

DaKingof commented:
The package is in the repo on the website, but it seems none of my tools on my system see's the package itself. Like it's registered but not connected properly somewhere. When attempting to rebuild my system, it says it's not available. The same goes when attempting to use nixenv. The only way it will install is by using `nix profile add` with the nixpkgs link and even then, I can only launch from the terminal, as no shortcut is created, and I would have to create one myself.
